### PR TITLE
64-bit VM Prover - Initial Tables

### PR DESCRIPTION
**Pull request by MauroToscano**
## 64-bit VM Prover - Initial Tables
  
  ### Summary

  Implements the foundation for the 64-bit VM prover using Goldilocks field with degree-3 extension (192-bit security).

  ### Changes

  **Core Types** (`prover/src/tables64/types.rs`)
  - Goldilocks prime field (p = 2^64 - 2^32 + 1) with Montgomery backend
  - Degree-3 cubic extension field (w³ = 2)
  - BusId enum for LogUp bus interactions

  **BITWISE Precomputed Table** (`prover/src/tables64/bitwise.rs`)
  - 2^20 rows (256 × 256 × 16) with 11 lookup types
  - Range checks: IS_BYTE, IS_HALF, IS_B20
  - Bitwise ops: AND_BYTE, OR_BYTE, XOR_BYTE, MSB8, MSB16, ZERO
  - Shift helpers: HWSL, HWSLC

  **Constraint Templates** (`prover/src/constraints64/templates.rs`)
  - `IsBitConstraint`: Enforces binary values via `cond * X * (1-X) = 0`
  - `AddConstraint`: 64-bit addition with embedded carry computation

  **QuadWL Packing** (`crypto/stark/src/lookup.rs`)
  - Added `QuadWL` variant for 4-word (128-bit) bus values

  ### Tests

  - Comprehensive edge case tests for BITWISE table (first/last row, MSB boundaries, shift boundaries)
  - Formula verification tests for IS_BIT and ADD carry computation
  - All 42 tests pass, clippy clean

### CODE REVIEW

File: prover/src/constraints64/templates.rs [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+    where
+        F: IsSubFieldOf<E>,
+        E: IsField,
+    {
+        let lhs_lo = step.get_main_evaluation_element(0, self.lhs_start_col);
+        let rhs_lo = step.get_main_evaluation_element(0, self.rhs_start_col);
+        let sum_lo = step.get_main_evaluation_element(0, self.sum_start_col);
+
+        // carry_0 = (lhs_lo + rhs_lo - sum_lo) * 2^(-32)
+        let inv_2_32: FieldElement<F> = FieldElement::from(SHIFT_32).inv().unwrap();
```
Comment (diegokingston) [inline prover/src/constraints64/templates.rs:459]: Just as a note, you can directly have FieldElement::from(SHIFT_32).inv().unwrap() this as a precomputed constant and avoid inversion at every step

File: prover/src/tests/bitwise_bus_tests.rs [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+// =============================================================================
+// Column indices (simplified for testing)
+// =============================================================================
+
+/// Sender table columns
+mod sender_cols {
+    pub const X: usize = 0;
+    pub const Y: usize = 1;
+    pub const AND_RESULT: usize = 2;
+    pub const MU_AND: usize = 3;
```
Comment (ColoCarletti) [inline prover/src/tests/bitwise_bus_tests.rs]: The name might be confusing; it should just be AND, since it works as a flag.

Comment (ColoCarletti) [inline prover/src/tests/bitwise_bus_tests.rs]:
```suggestion
    pub const AND: usize = 3;
```

File: prover/src/tests/bitwise_bus_tests.rs [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+        data[base + sender_cols::X] = FE::from(x as u64);
+        data[base + sender_cols::Y] = FE::from(y as u64);
+        data[base + sender_cols::AND_RESULT] = FE::from(result as u64);
+        data[base + sender_cols::MU_AND] = FE::one();
+    }
+
+    TraceTable::new_main(data, sender_cols::NUM_COLUMNS, 1)
+}
+
+fn create_receiver_trace(lookups: &[(u8, u8)]) -> TraceTable<F, E> {
```
Comment (ColoCarletti) [inline prover/src/tests/bitwise_bus_tests.rs]: To better emulate the behavior of a receiver table, it should have a single row per tuple (x, y, and), and increase the multiplicity when it receives more than one call.

**Review by ColoCarletti — APPROVED** [review]
